airbyte-integrations/connectors/source-mysql/src/main/kotlin/MySqlSourceExceptionHandler.kt

```diff
@@ -31,8 +31,73 @@ class MySqlSourceExceptionHandler : ConnectorExceptionHandler() {
                 failureType = FailureType.TRANSIENT,
                 externalMessage = "Can not read data from MySQL server",
                 sampleInternalMessage =
-                    "java.io.EOFException: Can not read response from server. Expected to read X bytes, read Y bytes before connection was unexpectedly lost."
-            )
+                    "java.io.EOFException: Can not read response from server. Expected to read X bytes, read Y bytes before connection was unexpectedly lost.",
+            ),
+        )
+
+        add(
+            ConnectorErrorProfile(
+                errorClass = "MySQL Hikari Connection Error",
```

Comment:
I interpreted this option to be matching on the classname of the exception that was thrown. But, I don't think that's what is happening here.  Is only the `regexMatchingPattern` used for finding the error to match?

Reply:
e.g. I would expect this to be `HikariPoolException` in this case

Reply:
I see; in this case, I think I did not interpret the `errorClass` correctly; I thought it was like a tag used to name the kind of errors...  so yes, currently, only regex is used, and the error message is still unique enough to get matched.

But yes, certainly, we can support this class as well. In your thoughts, this class should be a real type/class, but not a string, correct?

Reply:
and for Hikari errors like [this](https://github.com/airbytehq/oncall/issues/5346) (https://airbytehq.sentry.io/issues/5129832484/?project=6527718)

```
java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 10001ms
```

It does not have HikariPoolException, but `java.sql.SQLTransientConnectionException`, so we shall use this as the class I guess?

Reply:
I think if you are only using the regular expression match, and you want that match to be applied to all errors of all classes, I think you can skip the errorClass option.  Yes, I would epxect errorClass to be a real class or array of classess.

Reply:
That sounds good; I will submit another PR to implement this; this should not be very complicated to add to the existing translation mechanism.

Reply:
it's good you pointed out this.. i actually had difficulty explain what was the errorClass for.. now it makes much more sense...